### ISSUE TITLE
Ability to promote dead code analysis suggestion to warning or error

### ISSUE THREAD

**Issue by kendrahavens** [dotnet/roslyn#34357]
Ability to promote dead code analysis suggestion (IDE0051) to warning or error

![image](https://user-images.githubusercontent.com/8293542/54843595-fa9be380-4c91-11e9-85cb-436adcf768ae.png)

**Comment by Anduin2017** [comment]
Hi, @kendrahavens 

I have a question about how to detect useless code or functions. For sometimes I write an ASP.NET Core project and I just create a new action in my controller like this:

```csharp
    public class HomeController : Controller
    {
        public IActionResult Index()
        {
            return something;
        }

        public IActionResult Error()
        {
            return someOtherThing;
        }
    }
```

Of course, the `Index` method has zero references and seems to be dead and useless. But it wasn't because it was an action which makes sense.

If the compiler treats it as a `warning`, it will make my project whole of warnings. Any solution for that?

**Comment by kendrahavens** [comment]
Hey @Anduin2017,

What version are you seeing this warning? We did fix a bug where [interface-implementing events appeared unused](https://github.com/dotnet/roslyn/issues/32281). Fix was released in Dev16 Preview 2.

**Comment by Anduin2017** [comment]
@kendrahavens  Thanks for your response! I just worried that fixing this one may cause more issues for detecting more useful code useless.

**Comment by mavasani** [comment]
@kendrahavens You should be able to promote IDE0051 to a warning/error using a ruleset file entry. In future, when we add .editorconfig support in compiler to allow specifying diagnostic severity in .editorconfig file such that it is respected by the compiler, same functionality can be achieved by adding an entry in the .editorconfig file.

**Comment by mavasani** [comment]
@Anduin2017 The analyzer only flags unused _private_ members which are never referenced/invoked as dead code.

**Comment by mavasani** [comment]
@kendrahavens I believe once we have .editorconfig support in compiler, we should revive https://github.com/dotnet/roslyn/pull/29332 so we offer a code fix to configure severity of all reported diagnostics (including 3rd party analyzer diagnostics) via .editorconfig file

**Comment by ajeyjoshi** [comment]
@mavasani what shall be contents of the ruleset file "to promote dead code analysis suggestion (IDE0051) to warning or error"

**Comment by CyrusNajmabadi** [comment]
Closing out as this currently has editorconfig support.
